Add example using MV3 userScripts API (#576)
mdn/webextensions-examples: userScripts-mv3/README.md

# userScripts-mv3

A user script manager, demonstrating the userScripts API, the permissions API,
`optional_permissions`, and Manifest Version 3 (MV3).
The extension is an example of a
[user script manager](https://en.wikipedia.org/wiki/Userscript_manager).

> This covers the following aspects to extension development:

Author: I accepted Simeon's suggestion below instead, which adds "demo" to get to `This demo covers the following aspects to extension development:`

Is this wording OK or is it preferable to use "illustrate"?

> This covers the following aspects to extension development:

Reviewer: @Rob--W sorry for the delay two things probably wouldn't use the word demo and "the following" is wordy. "This example demonstrates these aspects of extension development."

- Showing onboarding UI after installation.

> - Designing background scripts that can restart repeatedly with minimal

Author: To make it easier to read when the file is not rendered.

# userScripts-mv3

The extension is an example of a
[user script manager](https://en.wikipedia.org/wiki/Userscript_manager). It
demonstrates the userScripts API, the permissions API, `optional_permissions`,
and Manifest Version 3 (MV3).

> This demo covers the following aspects to extension development:

Reviewer: ```suggestion
This example demonstrates these aspects of extension development:
```

should show this behavior:

- Show a dialog containing "This is a demo of a user script".
- Insert a button with the label "Show user script info", which opens a new tab
  displaying the extension information.

# What it shows

Showing onboarding UI after installation:

- `background.js` registers the `runtime.onInstalled` listener that calls
  `runtime.openOptionsPage` after installation.

Designing background scripts that can restart repeatedly with minimal overhead:

- This is particularly relevant to Manifest Version 3, because in MV3
  background scripts are always non-persistent and can suspend on inactivity.
- Using `storage.session` to store initialization status, to run expensive
  initialization only once per browser session.
- Registering events at the top level to handle events that are triggered while
  the background script is asleep.
- Using [dynamic import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import)
  to initialize optional JavaScript modules on demand.

Monitoring an optional (userScripts) permission, and dynamically registering
events and scripts based on its availability:

> - The `userScripts` permission is optional and can be granted by the user from:

Author: Optional-only is a subset of being an optional permission. Optional-only as a concept only makes a difference at install time; past install time the relevant user experience is identical to optional permissions.

It may be worth referring to the concept of optional-only permissions, but I'm not sure if the current section is the right place for that.
